File: Java/DailyLeetCode/maxValueAtEachLevel.java
class Solution {
    public List<Integer> largestValues(TreeNode root) {
        Queue<TreeNode> q= new LinkedList<>();
        List<Integer> ans= new ArrayList<>();
        if(root==null)
            return ans;

        q.add(root);
        while(!q.isEmpty()){
            int n= q.size();
            int max= Integer.MIN_VALUE;

            while(n-->0){
                TreeNode temp= q.poll();
                max= Math.max(max, temp.val);
                if(temp.left!=null) q.offer(temp.left);
                if(temp.right!=null) q.offer(temp.right);
            }

            ans.add(max);
        }

        return ans;
    }
}

File: Python/DailyLeetCode/maxValueAtEachLevel.py
class Solution:
    def largestValues(self, root: Optional[TreeNode]) -> List[int]:
        if not root:
            return []

        q = deque([root])
        ans = []

        while q:
            n = len(q)
            max_val = float('-inf')

            for _ in range(n):
                node = q.popleft()
                max_val = max(max_val, node.val)
                if node.left:
                    q.append(node.left)
                if node.right:
                    q.append(node.right)

            ans.append(max_val)

        return ans
